server/spawn-vehicle.lua

```diff
@@ -38,21 +70,105 @@ lib.callback.register('qbx_garages:server:spawnVehicle', function (source, vehic
         exports.qbx_core:Notify(source, locale('error.not_owned'), 'error')
         return
     end
-    if garageType == GarageType.DEPOT and FindPlateOnServer(playerVehicle.props.plate) then -- If depot, check if vehicle is not already spawned on the map
+
+    if garageType == GarageType.DEPOT and FindPlateOnServer(playerVehicle.props.plate) then
         return exports.qbx_core:Notify(source, locale('error.not_impound'), 'error', 5000)
     end
 
     local warpPed = Config.warpInVehicle and GetPlayerPed(source)
-    local netId, veh = qbx.spawnVehicle({ spawnSource = spawnCoords, model = playerVehicle.props.model, props = playerVehicle.props, warp = warpPed})
+    local netId, veh = qbx.spawnVehicle({
+        spawnSource = spawnCoords,
+        model = playerVehicle.props.model,
+        props = playerVehicle.props,
+        warp = warpPed
+    })
+
+    hookPayload.netId = netId
+    hookPayload.veh = veh
+
+    -- Trigger the spawnedVehicle hook
+    triggerEventHooks('garages:spawnedVehicle', hookPayload)
 
-    if Config.doorsLocked then
+    -- Use the doorsLocked value from the payload (which may have been modified by a hook)
+    if hookPayload.doorsLocked then
         SetVehicleDoorsLocked(veh, 2)
     end
```

Comment:
Rather than use the hook to give this info back to qbx_garages. The hook handler could do the vehicle locking itself, or even better, a vehicleSpawned event handler could do the locking and the config for qbx_garages doorsLocked would be set to false.

Reply:
For example, if the locking system is managed by something else, it can be deactivated.